python/nav/ipdevpoll/plugins/paloaltoarp.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (C) 2023 University of Tromsø
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License version 3 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""ipdevpoll plugin for fetching arp mappings from Palo Alto firewalls
+
+Add [paloaltoarp] section to ipdevpoll.conf
+add hostname = key to [paloaltoarp] section
+for example:
+[paloaltoarp]
+10.0.0.0 = abcdefghijklmnopqrstuvwxyz1234567890
+
+"""
+
+import xml.etree.ElementTree as ET
+from typing import Dict
+
+from IPy import IP
+from twisted.internet import defer, reactor, ssl
+from twisted.internet.defer import returnValue
+from twisted.web import client
+from twisted.web.client import Agent
+from twisted.web.http_headers import Headers
+
+from nav import buildconf
+from nav.ipdevpoll.plugins.arp import Arp
+
+
+class PaloaltoArp(Arp):
+    configured_devices: Dict[str, str] = {}
+
+    @classmethod
+    def on_plugin_load(cls):
+        """Loads the list of PaloAlto access keys from ipdevpoll.conf into the plugin
+        class instance, so that `can_handle` will be able to answer which devices
+        this plugin can run for.
+        """
+        from nav.ipdevpoll.config import ipdevpoll_conf
+
+        cls._logger.debug("loading paloaltoarp configuration")
+        if 'paloaltoarp' not in ipdevpoll_conf:
+            cls._logger.debug("PaloaltoArp config section NOT found")
+            return
+        cls._logger.debug("PaloaltoArp config section found")
+        cls.configured_devices = dict(ipdevpoll_conf['paloaltoarp'])
+
+    @classmethod
+    def can_handle(cls, netbox):
+        """Return True if this plugin can handle the given netbox."""
+        return (
+            netbox.sysname in cls.configured_devices
+            or str(netbox.ip) in cls.configured_devices
+        )
+
+    @defer.inlineCallbacks
+    def handle(self):
+        """Handle plugin business, return a deferred."""
+
+        api_key = self.configured_devices.get(str(self.netbox.ip), self.netbox.sysname)
```

Comment:
defaults API key to the sysname? or is this supposed to be like:
```suggestion
        api_key = self.configured_devices.get(str(self.netbox.ip)) or self.configured_devices.get(self.netbox.sysname)
```

Reply:
Indeed, you are correct :)